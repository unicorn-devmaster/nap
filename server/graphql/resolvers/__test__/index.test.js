/* eslint-env jest */
describe('index', () => {
  it('should install and authen then return authen', async () => {
    // stub
    global.NAP = {}
    NAP.Authen = {
      findOneAndUpdate: jest.fn().mockImplementationOnce(() =>
        Promise.resolve({ _id: '58d0e20e7ff032b39c2a9a18', name: 'bar' })
      )
    }

    NAP.Installation = {
      create: jest.fn().mockImplementationOnce(() =>
        Promise.resolve({ _id: '58d0e20e7ff032b39c2a9a19' })
      )
    }

    const { willInstallAndAuthen } = require('../index')
    const device = 'FOO_DEVICE'
    const user = { id: 'FOO_USER_ID' }
    const provider = 'facebook'

    const authen = await willInstallAndAuthen(device, user, provider)

    expect(authen).toMatchSnapshot()
  })
})