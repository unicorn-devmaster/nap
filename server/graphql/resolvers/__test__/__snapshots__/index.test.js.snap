// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index should install and authen then return authen 1`] = `
Object {
  "_id": "58d0e20e7ff032b39c2a9a18",
  "name": "bar",
}
`;
